apps/frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,16 @@
+import { Navigate, Outlet } from "react-router-dom";
+
+import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
+import { useAppSelector } from "~/libs/hooks/hooks.js";
+
+const ProtectedRoute: React.FC = () => {
```

Comment:
This component should receive `component` and `redirectTo` as props. You will either display that component or redirect to `redirectTo` screen